File: p14.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string ret = "";
        bool flag;
        char cmp;
        for(int i = 0; i < 200; i++){
            flag = true;
            cmp = strs[0][i];
            for(string s : strs){
                if(i == s.length() || s[i] != cmp) {
                    flag = false;
                    break;
                }
            }
            if(flag) ret += cmp;
            else break;
        }
        return ret;
    }
};

File: p14.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String ret = "";
        int mnLen = Integer.MAX_VALUE;
        int idx = 0;
        for(int i = 0; i < strs.length; i++){
            if(mnLen > strs[i].length()){
                mnLen = strs[i].length();
                idx = i;
            }
        }
        for(int i = 0; i < mnLen; i++){
            for(int j = 0; j < strs.length; j++){
                if(strs[j].charAt(i) != strs[idx].charAt(i)) return ret;
            }
            ret += strs[idx].charAt(i);
        }
        return ret;
    }
}